packages/providers/src/utils/types.ts

```diff
@@ -0,0 +1,14 @@
+export type Passthrough = {
+  body?: Record<string, unknown>;
+  headers?: Record<string, string>;
+  query?: Record<string, string>;
+};
+
+export type WithPassthrough<T> = Prettify<T & { _passthrough?: Passthrough }>;
+
+/**
+ * A type that represents a type that is a prettified version of the original type.
+ * The prettified type has all generics removed from intellisense and displays a flat object.
+ */
+// eslint-disable-next-line @typescript-eslint/ban-types
+export type Prettify<T> = { [K in keyof T]: T[K] } & {};
```

Comment:
<!-- Chores are simple tasks that must be done before the subject can be “officially” accepted. Usually, these comments reference some common process. Try to leave a link to the process description so that the reader knows how to resolve the chore.  -->
☑ **chore:** `Prettify` isn't necessary here, it's only useful for external/public SDK consumers to beautify the output. As this type is used internally by Providers, we don't need it.‏
```suggestion
export type WithPassthrough<T> = T & { _passthrough?: Passthrough };
```